### PR TITLE
Upgraded to Elasticsearch 2.0.0 and Kubernetes 1.1.1.

### LINKED ISSUE #23: Upgrade to Elasticsearch 2.0.0

### PR DESCRIPTION
**Pull request by pires**
@drora care to try this out?

Fixes #23

### CODE REVIEW

**Comment by drora** [comment]
Hi guys, gee that was quick..
I'll try to get up at it and let you know after the weekend.
Thanks a lot!

**Comment by pires** [comment]
@drora has been tested by me and other users and it's been merged to master. Let me know if you find any issues. Thank you.

**Comment by drora** [comment]
OK, tried it yesterday live, ended up with a rollback for 2 reasons:
1) tcp 9300 isn't reachable from the master node as it used to be.
2) http localhost:9200 isn't reachable anymore from client nodes.

**Comment by pires** [comment]
@drora can you detail the steps so I can reproduce?

**Comment by drora** [comment]
sure thing:
I spawned a cluster up as you describe in readme.md
after its up and running stable I sshd to a client instance and curl localhost:9200 (unreachable)
also tried to connect my java client via master node (tcp 9300) thru the master service and it couldn't establish connection (unreachable). rolling back to 1.7 solved all of the above...
cheers!

**Comment by pires** [comment]
@drora 

1. by `client` instance you mean `a pod exposing TCP 9200`, right?
2. Is your Java _client_ running inside the Kubernetes cluster? Can you provide more details?

**Comment by drora** [comment]
yes, client instance is a pod produced by the es-client-rc.
my java client is also running inside kube, it discovers master-service and reaching it via tcp 9300.
mind that java client was v2.0.0, and Elasticsearch migration plugin was green.
kube cluster is v1.1.1 aws based over ubunto running on t2.medium + 32gb ebs.
what else can be helpful?

**Comment by drora** [comment]
one more thing, I was able to access elasticsearch via es-svc on 9200.
the weird thing however, is if you ssh to a client pod exposing 9200 and try curl localhost:9200 it will not answer..

**Comment by pires** [comment]
I don't understand the concept of `ssh to a client pod`. There's no such thing as `ssh to a pod`, unless it exposes SSHd.

One more thing, TCP 9300 is used for cluster purposes alone. Applications that want to write/read data to/from the Elasticsearch cluster should use the HTTP API, hence TCP 9200.

**Comment by drora** [comment]
ok.
first, ssh to a pod with a single container -> ssh to a container, sorry for being unclear.
second, java client does support tcp connection via 9300 which is faster.
see: https://www.elastic.co/guide/en/elasticsearch/guide/current/_talking_to_elasticsearch.html
tl;dr: Java clients talk to the cluster over port 9300, using the native Elasticsearch transport protocol. The nodes in the cluster also communicate with each other over port 9300. If this port is not open, your nodes will not be able to form a cluster.

**Comment by pires** [comment]
So, TCP 9300 is not exposed with the `es-svc` (client nodes) but with `es-discovery-svc` (master nodes).

Again, are your pods exposing SSH? Or are you SSH'ing to a VM? On a VM, `localhost:9200` shouldn't be available and I wonder how you have with previous versions of this repo.

**Comment by drora** [comment]
currect, TCP 9300 is exposed with es-discovery-svc (master nodes). 
I had trouble connecting to them but that could be a problem on my side (client issue etc, didn't have time to profile the root cause and get into the buttom of this).
about the SSH:
I ssh to the elasticsearch client container (e.g. kubectl exec) and noticed that localhost:9200 doesn't answer as it used to in previous image.

**Comment by pires** [comment]
`kubectl exec` is not SSH.

I will look into it when I have some free time. Thank you for reporting.
